### PR TITLE
PYI-664 update template.yml with oauth client config

### PR DESCRIPTION
**Pull request by SunilDGDS**
### What changed

Configs have been created in order to authenticate OAuth clients in the system.  Two functions in core (Authorisation) &
token exchange to have access to configuration via env variables & permissions to param store

- [PYI-665](https://govukverify.atlassian.net/browse/PYI-665)

### CODE REVIEW

File: deploy/template.yaml [hunk 1]
```diff
@@ -64,11 +64,18 @@ Resources:
           ENVIRONMENT: !Sub "${Environment}"
           AUTH_CODES_TABLE_NAME: !Select [1, !Split ['/', !GetAtt AuthCodesTable.Arn]]
           ACCESS_TOKENS_TABLE_NAME: !Select [1, !Split ['/', !GetAtt AccessTokensTable.Arn]]
+          ORCHESTRATOR_ID: !Sub "/${Environment}/clients/orchestrator-stub/id"
+          ORCHESTRATOR_ISSUER: !Sub "/${Environment}/clients/orchestrator-stub/issuer"
+          ORCHESTRATOR_CERTIFICATE_FOR_CORE_TO_VERIFY: !Sub "/${Environment}/clients/orchestrator-stub/publicCertificateForCoreToVerify"
+          ORCHESTRATOR_VALID_CALLBACK_URL: !Sub "/${Environment}/clients/orchestrator-stub/validCallBackUrls"
+          ORCHESTRATOR_AUTHENTICATION_METHOD: !Sub "/${Environment}/clients/orchestrator-stub/authenticationMethod"
```
Comment (Wynndow) [inline deploy/template.yaml]: I think drop these for now. We'll probably just want to provide a path to the top level of the client config. Can do that in a later ticket though.

File: deploy/template.yaml [hunk 2]
```diff
@@ -92,11 +99,14 @@ Resources:
         Variables:
           ENVIRONMENT: !Sub "${Environment}"
           AUTH_CODES_TABLE_NAME: !Select [1, !Split ['/', !GetAtt AuthCodesTable.Arn]]
+          AUDIENCE_FOR_CLIENTS: !Sub "/${Environment}/self/audienceForClients"
```
Comment (Wynndow) [inline deploy/template.yaml]: As above

File: deploy/template.yaml [hunk 3]
```diff
@@ -64,11 +64,18 @@ Resources:
...
+          ORCHESTRATOR_CERTIFICATE_FOR_CORE_TO_VERIFY: !Sub "/${Environment}/clients/orchestrator-stub/publicCertificateForCoreToVerify"
+          ORCHESTRATOR_VALID_CALLBACK_URL: !Sub "/${Environment}/clients/orchestrator-stub/validCallBackUrls"
+          ORCHESTRATOR_AUTHENTICATION_METHOD: !Sub "/${Environment}/clients/orchestrator-stub/authenticationMethod"
       Policies:
         - DynamoDBCrudPolicy:
             TableName: !Ref AccessTokensTable
         - DynamoDBCrudPolicy:
             TableName: !Ref AuthCodesTable
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub ${Environment}/clients/orchestrator-stub/*
```
Comment (Wynndow) [inline deploy/template.yaml]:
```suggestion
            ParameterName: !Sub ${Environment}/clients/*
```

File: deploy/template.yaml (hunk 1 above)
Comment (SunilDGDS) [reply deploy/template.yaml]: removed

File: deploy/template.yaml (hunk 2 above)
Comment (SunilDGDS) [reply deploy/template.yaml]: removed

File: deploy/template.yaml (hunk 3 above)
Comment (SunilDGDS) [reply deploy/template.yaml]: updated

File: deploy/template.yaml [hunk 4]
```diff
@@ -69,6 +69,8 @@ Resources:
             TableName: !Ref AccessTokensTable
         - DynamoDBCrudPolicy:
             TableName: !Ref AuthCodesTable
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub ${Environment}/clients/*
```
Comment (Wynndow) [inline deploy/template.yaml]:
```suggestion
            ParameterName: !Sub ${Environment}/core/clients/*
```

File: deploy/template.yaml [hunk 5]
```diff
@@ -97,6 +99,8 @@ Resources:
             TableName: !Ref AuthCodesTable
         - DynamoDBReadPolicy:
             TableName: !Ref AuthCodesTable
+        - SSMParameterReadPolicy:
+              ParameterName: !Sub ${Environment}/self/audienceForClients
```
Comment (Wynndow) [inline deploy/template.yaml]:
```suggestion
              ParameterName: !Sub ${Environment}/core/self/audienceForClients
```

File: deploy/template.yaml (hunk 4 above)
Comment (SunilDGDS) [reply deploy/template.yaml]: added missing core

File: deploy/template.yaml (hunk 5 above)
Comment (SunilDGDS) [reply deploy/template.yaml]: added missing core

**Review by Wynndow — APPROVED** [review]

**Pull request merged**